src/util/crypto_browser.ts

```diff
@@ -0,0 +1,135 @@
+// According to their type declarations, SubtleCrypto functions return `PromiseLike`.
+// They are close enough to a `Promise` to `await`, but tslint doesn't know that.
+/* tslint:disable:await-promise */
+
+const { crypto } = window
+const HASH_ALGORITHM = 'SHA-256'
+const ENCRYPTION_ALGORITHM = 'AES-GCM'
+const IV_LENGTH = 12
+const AUTH_TAG_BYTES = 16
+const AUTH_TAG_BITS = 8 * AUTH_TAG_BYTES
+const CACHE_EXPIRY = 30000
+
+// Cache keys so that `subtle.importKey` doesn't need to be called for every operation.
+// It would be nicer to just store the `CryptoKey`s on the stream `Connection`, but
+// that's tricky since this file takes the place of `crypto_node.ts`.
+class KeyCache {
+  private cache: Map<Buffer, CacheEntry> = new Map()
+
+  cleanup () {
+    const now = Date.now()
+    for (const entry of this.cache) {
```

Comment:
This can be cleaner by destructing the entry array inline
`for (const [key, value] of this.cache)`